src/libcore/hint.rs

```diff
@@ -104,11 +104,28 @@ pub fn spin_loop() {
     }
 }
 
-/// A function that is opaque to the optimizer, to allow benchmarks to
-/// pretend to use outputs to assist in avoiding dead-code
-/// elimination.
+/// An identity function whose argument is opaque to the optimizer. This is useful for (e.g.)
```

Comment:
I believe this is incorrect, the argument isn't opaque to the optimizer, and optimizations on the argument are possible, e.g., `black_box(2 * 5)` is currently optimized to `black_box(10)`.

Reply:
Also, "is opaque" is also incorrect - this function is a `hint`, there are no guarantees that it will do anything at all.